eng/scripts/docs/tests/Toc-Generation-Tests.ps1

```diff
@@ -0,0 +1,115 @@
+<#
+```
+Invoke-Pester -Output Detailed $PSScriptRoot\Toc-Generation-Tests.ps1
+```
+#>
+
+Import-Module Pester
+
+BeforeAll {
+    . $PSScriptRoot/../Docs-ToC.ps1
+    function Backup-File($targetPath, $backupFolder) {
+        if (!(Test-Path $targetPath)) {
+            return $null
+        }
+        $fileName = (Split-Path $targetPath -leaf)
+        $backupFile = "$backupFolder/temp-$fileName"
+        $null = New-Item $backupFile -ItemType "file" -Force
+        $null = Copy-Item $targetPath -Destination $backupFile
+        return $backupFile
+    }
+    function Reset-File($targetPath, $backupFile) {
+        if ($backupFile) {
+            $null = Copy-Item $backupFile -Destination $targetPath
+        }
+    }
+}
+
+AfterAll {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "extractNupkg")
+    Remove-Item "$tempLocation/*" -Recurse
+    Remove-Item "$PSScriptRoot/outputs" -Recurse
+}
+# Test plan:
+# 1. Tests on Fetch-NamespacesFromNupkg from nuget source. 
+# 2. Tests on Fetch-NamespacesFromNupkg from public feeds. 
+# 3. Tests on Get-Toc-Children for latest
+# 4. Tests on Get-Toc-Children for preview
+Describe "Fetch-NamespacesFromNupkg-Nuget" -Tag "UnitTest" {
+    # Passed cases
+    It "Fetch namespaces from package downloads from nuget" -TestCases @(
```

Comment:
We should have the cosmos packages that were causing issues added as test cases.

Reply:
If cosmos has been problematic, it should definitely be added to your list of test cases, shouldn't it?

Reply:
This is newly found case. Will add to test.

Reply:
I will use Azure.Cosmos one for testing as this is deterministic. `Microsoft.Azure.Cosmos` will pick the first one which is flaky from test perspective.

Reply:
If we think Microsoft.Azure.Cosmos is flaky that means that we also have a flaky implementation that will give indeterministic results. If that is the case we should fix the implementation to order the dlls so that the result is more stable.

Reply:
Then we need to order the dll first and then fetch the first.

Reply:
Done with the change